### ISSUE TITLE
Question: using eager fields in query definition

### ISSUE THREAD

**Issue by casalot** [Vincit/objection.js#186]
Is there a way to use fields from the eager query in `where` and/or `orderBy`?

Example:
```javascript
    Incident
      .query()
      .eager('[incidentStatus]')
      .where('incidentStatus.sortOrder','>=','0')
```
When using the example above, an error is thrown about the `incidentStatus.sortOrder` field that does not exist.

**Comment by fl0w** [comment]
I think you're looking for [filterEager](http://vincit.github.io/objection.js/#filtereager).
Since objection doesn't do a join-matrix for eager loading relations, it makes sense that `incidentStatus` fileds are inaccessible in the main query.

**Comment by koskimas** [comment]
Not with `eager`. See the conversation in issue #73. You can use `joinRelation` for single (not nested) relations like the example:

```js
    Incident
      .query()
      .select('Incident.*')
      .joinRelation('incidentStatus')
      .where('incidentStatus.sortOrder','>=','0')
```

**Comment by koskimas** [comment]
This issue is addressed in #73.
